Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java

```diff
@@ -1,61 +1,206 @@
 package org.firstinspires.ftc.team6220_PowerPlay;
 
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
+import org.firstinspires.ftc.robotcore.external.navigation.AxesReference;
+
 public abstract class BaseTeleOp extends BaseOpMode {
+    double currentAngle;
+    double headingDegrees;
+    double negativeHeadingRadians;
+
+    double x;
+    double y;
+    double t;
+
+    double xRotatedVector;
+    double yRotatedVector;
+
+    double angleToCardinal;
+    double ratio;
+
     double xPower;
     double yPower;
     double tPower;
 
-    public void driveChassisWithController() {
-        xPower = gamepad1.left_stick_x * Constants.DRIVE_SPEED_MULTIPLIER * (1 - gamepad1.left_trigger * 0.5);
-        yPower = gamepad1.left_stick_y * Constants.DRIVE_SPEED_MULTIPLIER * (1 - gamepad1.left_trigger * 0.5);
-        tPower = gamepad1.right_stick_x * Constants.DRIVE_SPEED_MULTIPLIER * (1 - gamepad1.left_trigger * 0.5);
+    int slideTargetPosition = 0;
+
+    int stack = 0;
+    int[] stacks = {0, 0};
```

Comment:
What are these stacks?

Reply:
it signifies the number of cones on the autonomous stack

Reply:
Can you add a comment in the code saying that? Usually if you have to clarify something to a reviewer, you or someone else new to your team will also need that clarification later on. The code should be clear by itself without looking at the pull request comments.

Also why is there a `stack` variable and a `stacks` variable? I guess similar question for `junction` and `junctions`. It's not clear what these variables are for.

Reply:
I added comments for this, basically the stack is for the cone stack and junction is for the junction
The stacks is to keep track of current and previous stack values and the junctions is to keep track of current and previous junction values

Reply:
Your comment in the code needs to be at least as detailed as your comment here and I don't think your comment here is making this clearer for me. Why do you need to keep track of the current and previous stack values? Wouldn't you always have one less cone on the stack after you remove a cone? What do current and previous junction values even mean in this scenario? Aren't you always going for the high junction?